sycl/source/detail/event_impl.hpp

```diff
@@ -50,7 +50,13 @@ class event_impl {
   /// a device event.
   event_impl(std::optional<HostEventState> State = HES_Complete)
       : MIsInitialized(false), MHostEvent(State), MIsFlushed(true),
-        MState(State.value_or(HES_Complete)) {}
+        MState(State.value_or(HES_Complete)) {
+    // Need to fail in event() constructor  if there are problems with the
+    // ONEAPI_DEVICE_SELECTOR. Deferring may lead to conficts with noexcept
+    // event methods. This ::get() call uses static vars to read and parse the
+    // ODS env var exactly once.
+    SYCLConfig<ONEAPI_DEVICE_SELECTOR>::get();
+  }
```

Comment:
Is there a reason we do it in the ctor of `event_impl` and not `event`?

Reply:
Eventually I would like us to try instantiating the event's context when we know the event is wanted by the user, i.e. keep `event_impl` as long as we can, then once it becomes owned by an `event` it will instantiate the context. Alas, for now, I am alright with this solution, assuming it is relatively cheap.